src/V3Ast.cpp

```diff
@@ -1141,6 +1141,51 @@ void AstNode::dumpTreeFile(const string& filename, bool append, bool doDump, boo
     }
 }
 
+static std::string nodeAddr(const AstNode* nodep) {
+        return v3Global.opt.dumpTreeAddrids() ? v3Global.ptrToId(nodep) : cvtToHex(nodep);
+}
+
+static void drawEdge(std::ostream& os, const AstNode* n1,  const AstNode* n2, const std::string& name) {
+    if (n2) {
+        os << "\tn" << cvtToHex(n1) << " -> n" << cvtToHex(n2) << " ["
+           << "label=\"" << name << "\" color=red];\n";
+    }
+    for (const AstNode* nodep = n2; nodep; nodep = nodep->nextp()) {
+        nodep->dumpTreeDot(os);
+        if (nodep->nextp()) {
+            os << "\tn" << cvtToHex(nodep) << " -> n" << cvtToHex(nodep->nextp()) << " ["
+               << "label=\"next\" color=red];\n";
+            os << "\t{rank=same; n" << cvtToHex(nodep) << ", n" << cvtToHex(nodep->nextp())
+               << "}\n";
+        }
+    }
+}
+
+void AstNode::dumpTreeDot(std::ostream& os) const {
+    os << "\tn" << nodeAddr(this) << "\t["
```

Comment:
This should be just `cvtToHex(nodep)` (which is what you use everywhere else) instead of `nodeAddr(this)`